### DISCUSSION TITLE
Dynamic audio sample rate.

### DISCUSSION THREAD

**Discussion by aleph1** [Kode/Kore#586] [Ideas]
I am trying to wrap my head around whether it is possible to get an updated audio sample rate as this can technically change on application launch on OS X (don't use Windows or Linux, but wonder if they have the same issue). It looks like sample rate is set on kinc_a2_init here: https://github.com/Kode/Kinc/blob/master/Backends/System/macOS/Sources/Kore/Audio.cpp

However, on the Mac you can change output audio format dynamically for the OS using the “Audio MIDI Setup” application. I'm only encountering this issue as I am working on a game project that allows for dynamic audio to be passed out using a MIDI signal, and it seems fairly specific. I would like to add a query to Kinc that allows checking the current sample rate each time the audio buffer is expecting to be filled. Is there a specific way you see an interface like this working?

**Comment by RobDangerous** [comment]
Hey, looks like the iOS backend already supports that, no interface changes needed.

**Comment by aleph1** [comment]
I feel like I am missing something. The sample rate seems to be established in kinc_a2_init, and kinc_a2_samples_per_second seems to be only set once. Can I just call kinc_a2_init again when a buffer is expected to fill to get the current sample rate? Being named *init I expected it to be only called once. Again, I'm interested in both the sample rate when the application launches, but also the sample rate while the application is running, which can change. Thanks!

**Comment by RobDangerous** [comment]
No no, do it just like the iOS backend does.
